### PR TITLE
Use strongly connected components of directed relations graph

### PR DESCRIPTION
**Pull request by tilk**
This PR replaces searching for connected components of the undirected relations graph with strongly connected components of the directed relations graph (undirected conflicts, directed priorities). This change makes the components smaller in some cases, which might become handy in the future.

### CODE REVIEW

File: coreblocks/transactions/_utils.py [hunk 1]
```diff
@@ -116,6 +117,63 @@ def _graph_ccs(gr: ROGraph[T]) -> list[GraphCC[T]]:
...
+    Parameters
+    ----------
+    gr : Mapping[T, Iterable[T]]
+        Graph in which we should find strongly connected components. Encoded
+        using adjacency lists.
+
+    Returns
+    -------
+    ccs : List[Set[T]]
+        Connected components of the graph `gr`.
```
Comment (piotro888) [inline coreblocks/transactions/_utils.py:134]: Old function name in first line of docstring.
Old return var name and missing "strongly"?

**Review by piotro888 — APPROVED** [review]
LGTM, only slight docstring issues

**Comment by tilk** [comment]
Converted to draft, as I need to think some things through.

**Comment by tilk** [comment]
Closing to unclutter the pull request list a bit.

**Pull request closed without merging**